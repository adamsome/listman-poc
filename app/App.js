import React from 'react'
import { Link } from 'react-router'

import Footer from './components/Layout/Footer'
import BodyColumn from './components/Layout/BodyColumn'
import Columns from './components/Layout/Columns'
import Column from './components/Layout/Columns/Column'
import './App.scss'

export default class App extends React.Component {
  render() {
    const { children } = this.props
    return (
      <div>
        <section className="section">
          {children}
        </section>
        <Footer>
          <BodyColumn>
            {/* Temp dev tools */}
            <h4 className="title is-4">Dev tools</h4>
            <Columns multiline>
              <Column>
                <strong>User links</strong>
                <br />
                <Link to='/adamsome'>/adamsome</Link>
                <br />
                <Link to='/other-user'>/other-user</Link>
                <br />
                <Link to='/not-a-user'>/not-a-user</Link>
              </Column>
            </Columns>
          </BodyColumn>
        </Footer>
      </div>
    )
  }
}
