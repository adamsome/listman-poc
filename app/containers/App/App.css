.App {
}

.App-header {
  display: flex;
  border-bottom: 1px solid #000;
}

.App-intro {
  font-size: large;
}

