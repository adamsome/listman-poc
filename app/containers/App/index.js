import React from 'react'

import './App.css'
import List from '../List'

export default class App extends React.Component {
  render() {
    return (
      <div className="App">
        <h1 className="App-header">
          Basic
        </h1>
        <List />
      </div>
    )
  }
}
