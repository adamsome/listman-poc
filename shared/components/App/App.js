import React from 'react'
import { Link, Match, Miss } from 'react-router'

import AsyncHome from '../DemoApp/AsyncHome';
import AsyncPosts from '../DemoApp/AsyncPosts';
import Error404 from '../DemoApp/Error404';

import Footer from '../Layout/Footer'
import BodyColumn from '../Layout/BodyColumn'
import Columns from '../Layout/Columns'
import Column from '../Layout/Columns/Column'

import './App.scss';

const App = () => (
  <div>
    <section className="section">
      <Match exactly pattern="/" component={AsyncHome} />
      <Match pattern="/posts" component={AsyncPosts} />
      <Miss component={Error404} />
    </section>
    <Footer>
      <BodyColumn>
        {/* Temp dev tools */}
        <h4 className="title is-4">Dev tools</h4>
        <Columns multiline>
          <Column>
            <strong>Temp</strong>
            <br />
            <Link to='/'>/[home]</Link>
            <br />
            <Link to='/posts'>/posts</Link>
          </Column>
          <Column>
            <strong>User links</strong>
            <br />
            <Link to='/adamsome'>/adamsome</Link>
            <br />
            <Link to='/other-user'>/other-user</Link>
            <br />
            <Link to='/long-maximum-length-user-name-32'>
              /long-maximum-length-user-name-32
            </Link>
            <br />
            <Link to='/not-a-user'>/not-a-user</Link>
          </Column>
        </Columns>
      </BodyColumn>
    </Footer>
  </div>
)

export default App
