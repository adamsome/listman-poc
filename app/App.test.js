import React from 'react'
import { shallow } from 'enzyme'

import App from './App'

describe('<App>', () => {
  it('exists', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toBeDefined()
  })

  it('renders as a <div>', () => {
    const wrapper = shallow(<App />)
    expect(wrapper.type()).toEqual('div')
  })
})
