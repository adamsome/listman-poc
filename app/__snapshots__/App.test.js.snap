exports[`test renders correctly 1`] = `
<div>
  <section
    className="section" />
  <Footer>
    <MainColumns>
      <h4
        className="title is-4">
        Dev tools
      </h4>
      <Columns
        multiline={true}>
        <Column>
          <strong>
            User links
          </strong>
          <br />
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/adamsome">
            /adamsome
          </Link>
          <br />
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/other-user">
            /other-user
          </Link>
          <br />
          <Link
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/not-a-user">
            /not-a-user
          </Link>
        </Column>
      </Columns>
    </MainColumns>
  </Footer>
</div>
`;
